connection.ts

```diff
@@ -219,7 +221,7 @@ export class Connection {
   private async _authMd5(salt: Uint8Array) {
     this.packetWriter.clear();
     const password = hashMd5Password(
-      this.connParams.password,
+      this.connParams.password!,
```

Comment:
this is the only type assertion that I added that I felt really uncomfortable with, because I'm not sure if this ever gets called with a null password, and what the implications of that are.

Reply:
It might be possible that `password` in this case is null. If you can please check what message postgres returns in case of `null` password - if it's too cryptic you can add an assertion in `_authMd5` before sending message to Postgres

Reply:
the _authMd5 doesn't work in the case of connParams.password being unset, so it probably make sense to throw a guard clause above it that throws an error if it's unset. That'll make the type assertion unnecessary.